drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -1832,10 +1834,16 @@ static int vc4_hdmi_encoder_atomic_check(struct drm_encoder *encoder,
 		pixel_rate = mode->clock * 1000;
 	}
 
+
 	ret = vc4_hdmi_encoder_compute_config(vc4_hdmi, vc4_state, mode);
 	if (ret)
 		return ret;
 
+	/* vc4_hdmi_encoder_compute_config may have changed output_bpc and/or output_format */
+	if (vc4_state->output_bpc != output_bpc ||
+	    vc4_state->output_format != output_format)
```

Comment:
And then here you can just compare `vc4_state->output_bpc != old_vc4_state->output_bpc`.

That way you don't rely on the fact that nothing updates output_bpc in the current state before that code runs, you always consider the old state value.